src/components/workspace-utils.js

```diff
@@ -131,14 +132,18 @@ export const SnapshotInfo = ({
   // Lifecycle
   useOnMount(() => {
     const loadSnapshotInfo = async () => {
-      const snapshotInfo = await Ajax(signal).DataRepo.snapshot(snapshotId).details()
-      setSelectedSnapshotInfo(snapshotInfo)
+      try {
+        const snapshotInfo = await Ajax(signal).DataRepo.snapshot(snapshotId).details()
+        setSelectedSnapshotInfo(snapshotInfo)
+        setSnapshotLoadError(undefined)
+      } catch (e) {
+        setSelectedSnapshotInfo({ createdDate: Date.now() }) // fake data prevents problems with Utils.makeCompleteDate below
```

Comment:
The only place I see `createdDate` used is in a block that's conditional on `!snapshotLoadError`. If these two lines were switched, so that `setSnapshotLoadError(e)` is called before `setSelectedSnapshotInfo`, is the fake data still necessary?

Reply:
ahh, good call! 🤯 changed, and that works much better. Thank you.

Reply:
I think such problems will go away with [React 18's automatic batching](https://reactjs.org/blog/2022/03/29/react-v18.html#new-feature-automatic-batching).